CommandLineParser/Core/Usage/UsageBuilder.cs

```diff
@@ -79,10 +79,10 @@ public void PrintOptions(IEnumerable<ICommandLineOption> options, int descriptio
 
             stringBuilder.AppendLine().AppendLine("Options: ");
 
-            var longestOptionName = options.Max(x => x.ShortName.Length + x.LongName.Length);
-            var separatorCompensation = options.Any(x => x.HasShortName && x.HasLongName) ? 1 : 0;
+            var longestOptionName = options.Max(x => (x.HasShortName ? x.ShortName.Length : 0) + (x.HasLongName ? x.LongName.Length : 0));
+            var compensateSeparator = options.Any(x => x.HasShortName && x.HasLongName);
             foreach (var opt in options)
-                PrintOption(opt, longestOptionName - opt.LongName.Length - opt.ShortName.Length + separatorCompensation + descriptionShift);
+                PrintOption(opt, longestOptionName - (opt.HasLongName ? opt.LongName.Length : 0) - (opt.HasShortName ? opt.ShortName.Length : 0) + descriptionShift, compensateSeparator);
```

Comment:
The `PrintOption` is also getting pretty big, might be better to split it into some variables.